lib/THC/THCTensorIndex.cu

```diff
@@ -173,6 +176,31 @@ void THCudaTensor_indexFill(THCState *state, THCudaTensor *res_, int dim, THCuda
   );
 
   THCudaCheck(cudaFree(stride_));
+  THCudaTensor_free(state, indices);
+}
+
+__global__ void THCudaTensor_kernel_indexSelect_contiguous(
+  float *tensor, float *src, long stride, float *index, long idxSize)
+{
+  // In the typical case, each block of 128 threads handles a 4x128
+  // section of the output with each warp handling a single 1x128 row.
+  // The outer loops handle inputs larger than 4*65535 or strides larger
+  // than 128*65535.
+
+  for (int idx = blockIdx.x * 4 + threadIdx.y; idx < idxSize; idx += 4*65535) {
```

Comment:
Can you use `blockDim.x` etc instead of constants? It makes the code more easily readable and less error-prone.